packages/core/ui-settings/core-ui-settings-server-internal/src/settings/theme.ts

```diff
@@ -46,12 +46,31 @@ export const getThemeSettings = (
       name: i18n.translate('core.ui_settings.params.darkModeTitle', {
         defaultMessage: 'Dark mode',
       }),
-      value: defaultDarkMode,
+      value: defaultDarkMode ? 'enabled' : 'disabled',
       description: i18n.translate('core.ui_settings.params.darkModeText', {
         defaultMessage: `Enable a dark mode for the Kibana UI. A page refresh is required for the setting to be applied.`,
       }),
+      type: 'select',
+      options: ['enabled', 'disabled', 'system'],
+      optionLabels: {
+        enabled: i18n.translate('core.ui_settings.params.darkMode.options.enabled', {
+          defaultMessage: `Enabled`,
+        }),
+        disabled: i18n.translate('core.ui_settings.params.darkMode.options.disabled', {
+          defaultMessage: `Disabled`,
+        }),
+        system: i18n.translate('core.ui_settings.params.darkMode.options.system', {
+          defaultMessage: `Sync with system`,
+        }),
+      },
       requiresPageReload: true,
-      schema: schema.boolean(),
+      schema: schema.oneOf([
+        schema.literal('enabled'),
+        schema.literal('disabled'),
+        schema.literal('system'),
+        // for backward-compatibility
+        schema.boolean(),
+      ]),
```

Comment:
I had to change the type of field because of that third option, and you can't mix boolean and string values, so I had to use strings for all values.

However I kept the possibility for users to still set a boolean value, and I added an utility function (`parseDarkModeValue`) to sanitize the value that could be stored. That way, we're upward compatible, and we're downward compatible as long as the value is not overridden from a new node (which seems more than good enough given what we know of serverless now).

Also worse case (new node sets the value and old node reads it), it just causes the `dark` mode to be used because the string values will be coerced to boolean's `true`.